packages/shadydom/src/patches/ElementOrShadowRoot.js

```diff
@@ -37,6 +37,16 @@ export const ElementOrShadowRootPatches = utils.getOwnPropertyDescriptors({
       this[utils.NATIVE_PREFIX + 'innerHTML'] = value;
     } else {
       clearNode(this);
+      // Make sure that the domain attributes of inertDoc and the document for
+      // are the same in order to avoid WrongDocumentError in IE11.
+      if ((inertDoc.domain.indexOf(document.domain) >= 0) && (inertDoc.domain !== document.domain)) {
```

Comment:
Doing this check on every call to `innerHTML` is likely bad for performance. It would be better to override the setter for `domain` on document and sync the inertDoc domain there.